bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/internal/SitemapResource.java

```diff
@@ -714,7 +803,7 @@ private boolean blockUnlessChangeOccurs(String sitemapname, String pageId) {
      *            the widgets of the page to observe
      * @return true if the timeout is reached
      */
-    private boolean waitForChanges(EList<Widget> widgets) {
+    private boolean waitForChanges(List<Widget> widgets) {
         long startTime = (new Date()).getTime();
         boolean timeout = false;
         Set<String> items = getAllItems(widgets).stream().map(Item::getName).collect(Collectors.toSet());
```

Comment:
This will be interesting that you add a DEBUG log just after this line showing how many widgets you have and how many items it resulted.

Reply:
In a common user situation (user defining only one sitemap), it should lead to have more or less all items.

Reply:
Good idea, I added that but in the method in SitemapSubscriptionService and deduplicated the code from this method.

Reply:
Sorry, I got that wrong, added the log statement into waitForChanges as well.